modules/views/components/civicrm.core.inc

```diff
@@ -2711,9 +2711,10 @@ function _civicrm_core_data(&$data, $enabled) {
 
 
   while ($dao->fetch()) {
-    // call getTree using $dao->id as groupID and $dao->extends as entityType
-
-    $data = civicrm_views_custom_data_cache($data, $dao->extends, $dao->id, $dao->extends_entity_column_value);
+    // call getTree using $dao->id as groupID, $dao->extends as entityType, with possible subtypes in $dao->extends_entity_column_value
+    $contact_subtypes = (('Contact' != $dao->extends) || empty($dao->extends_entity_column_value)) ? NULL : array_filter(explode(CRM_Core_DAO::VALUE_SEPARATOR, $dao->extends_entity_column_value));
```

Comment:
There's a lot happening on this line! This is a bit easier to read/maintain:

```php
    $extendsContactSubtype = (('Contact' === $dao->extends) && !empty($dao->extends_entity_column_value));
    $contact_subtypes = !$extendsContactSubtype ? NULL : array_filter(explode(CRM_Core_DAO::VALUE_SEPARATOR, $dao->extends_entity_column_value));
```

Also, I think civilint will reject `$contact_subtypes`. `$contactSubtypes` should work, though.